cmd/thanos/flags.go

```diff
@@ -145,6 +145,31 @@ func modelDuration(flags *kingpin.FlagClause) *model.Duration {
 	return value
 }
 
+type flagTime struct {
```

Comment:
so I belive this is quite fixed... We essentially need duration right?  Like `now-3months` - `now-2h` style.

Reply:
Yeah I agree I need relative time also, would essentially replace the functionality I got here https://github.com/improbable-eng/thanos/pull/930

Reply:
Yes, that would be amazing!

Reply:
I think both would be useful.  We're intending to use this for horizontal scale out, and we want the ability to specify exact time ranges in order to size the blocks served to the capacity of the thanos store host.

Reply:
:+1: I think both are useful

Reply:
@GiedriusS you are fine with this not being in Thanos Query?  :)

Reply:
@povilasv I mean if this will support relative times then it would only be a side-effect that you can do something like in #930 and, again, I really doubt that someone will quickly scramble all of their deployment scripts to redeploy Thanos Store nodes with a different option to see all of their data in remote storage if all of their Thanos Sidecar nodes go down. Especially considering how damn long it takes for them to start up :smile:  I just didn't want to add an option like that to Thanos Store if we could do the same thing in a general way like in this pull request. Wouldn't it be much more elegant and useful? 
All in all - yes, I would be fine with that. Also, Thanos Query could still support such thing that we've talked about. Sorry if I've seemed a bit aggressive in the comments, I don't have any hard feelings :smiley: I love this idea in general

Reply:
Nah no hard feelings, just wanted to get the consensus going :D

Reply:
:+1: 

One question: I don't get why someone would need specific time. I think it does not make sense. What's the use case?

> We're intending to use this for horizontal scale out, and we want the ability to specify exact time ranges in order to size the blocks served to the capacity of the thanos store host.

@claytono hmm are you sure you want fixed time ranges for that? Why?

Reply:
We intend to have a handful of Thanos store nodes, each serving a portion of a shared bucket, with one of them having an open ended time range for all new metrics.  For now, we intend to have an outside process periodically do analysis of the bucket and generate time ranges for each thanos store process based on index size for each block.  We want to provision these nodes such that they're all fairly full from a memory standpoint, but that we're not over-provisioning.  For us, the major expense of running Thanos is the memory on compute instances, and the S3 storage is nearly free in comparison.

With metric ingest rates changing over time (new apps, seasonality, etc) and the activity of the compactor, I think partitioning the bucket time ranges with relative times is going to be error prone and/or lead to inefficient usage of the hardware.

Reply:
It's kind of odd from my perspective, but if you find this useful, sure (: happy to accept that.

Reply:
It's what we've come up with for horizontal scaling of the Thanos store nodes.  I'd love to hear how other people are managing scaling out.

Reply:
>For us, the major expense of running Thanos is the memory on compute instances
Seems, you are trying to solve separate problem with absolute time ranges.

I'd like to have relative time as well.